jwt-bcp/draft-ietf-oauth-jwt-bcp.md

```diff
@@ -334,6 +347,8 @@ the Password Considerations in Section 8.8 of [RFC7518]
 MUST be followed.
 In particular, human-memorizable passwords MUST NOT be directly used
 as the key to a keyed-MAC algorithm such as "HS256".
+Rather, the principles from {{RFC2898}} SHOULD be used
+to derive cryptographic keys from user-supplied passwords.
```

Comment:
I strongly disagree with the last sentence. Even with PBKDF applied, this is still vulnerable to brute force attacks. PBKDF is useful when you have a bunch of passwords that you want to protect as an aggregate. Not when you have a single high-value JWT being attacked.

To put the above into perspective: are you aware of any real use cases where people must use human passwords to sign JWTs?

Reply:
Key Encryption with PBES2 was included in JWA at https://tools.ietf.org/html/rfc7518#section-4.8 because of demand for it in the JOSE working group.  I could search the list archives to find out who and maybe why, if you're really interested.

JWA already references using the principles in RFC 2898 when using these algorithms.  This added sentence makes that existing requirement visible in the BCP as well.

We're not recommending using passwords. But Ekr wanted us to clarify the statement about directly using passwords to encrypt content.  Thinking about it some more, I think we should delete the new sentence I wrote and replace it with this:

"Rather, passwords should only be used to perform key encryption, rather than content encryption, as described in Section 4.8 of [RFC 7518]."

Would that sentence work for you?

Reply:
Note that still vulnerable to brute force attacks

Reply:
Also added:

Note that even when use for key encryption, password-based encryption is still subject to brute-force attacks.